include/networkit/auxiliary/Timer.hpp

```diff
@@ -102,6 +102,36 @@ class StartedTimer : public Timer {
 	}
 };
 
+/**
+ * A timer for running time measurements within a scope.
+ *
+ * Same as Timer but automatically starts on construction and report on destruction.
+ *
+ * @code
+ * {
+ *    Aux::ScopedTimer someName("Algorithm B");
+ *    // some expensive operations
+ *
+ *
+ * } // <- Report time since creation of the timer object to std::cout
+ * @endcode
+ *
+ * @warning Similarly as std::unique_lock the timer needs a name ("someName" in the example)
+ * to exist until the end of the scope. If the time measured seems too short, make you created
+ * a named instances.
+ */
+class ScopedTimer : public StartedTimer {
```

Comment:
This class has no users. Do we need it? When / how will users be added?

Reply:
I consider this a developing tool. I typically have a copy of this class in my source tree and remove it before pushing.

Reply:
In this case, would you consider renaming it to something like `DebuggingTimer` / `DevelopmentTimer` / `LoggingTimer` (or some bikesheded version of that)?